crates/bevy_color/src/hsla.rs

```diff
@@ -66,6 +66,24 @@ impl Hsla {
     pub const fn with_lightness(self, lightness: f32) -> Self {
         Self { lightness, ..self }
     }
+
+    /// Generate a deterministic but randomly distributed color from a provided `index`.
+    /// This can be helpful for generating debug colors.
+    pub fn from_index(index: u32) -> Self {
```

Comment:
I think we can come up with a better name here. Maybe "generate_dispersed"?

Reply:
On this idea, I've decided to instead off a function `sequence_dispersed` which returns an iterator of these dispersed colours. Since it's backed by a `Range<u32>`, the `nth` method is `O(1)` (as fast as the original method) whilst also giving some nice flexibility in its use (e.g., `take(5)`, `for color in Hsla::sequence_dispersed() { .. }`, etc.)